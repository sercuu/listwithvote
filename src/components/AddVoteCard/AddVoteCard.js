/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { Icon, Button, Form, Input } from 'semantic-ui-react';

export const AddVoteCardContainer = () => {
  const [name, setName] = useState('');
  const [path, setPath] = useState('');

  const handleNameChange = e => {
    setName(e.target.value);
  };

  const handlePathChange = e => {
    setPath(e.target.value);
  };

  const handleSubmit = () => {
    const localStorageVoteList = JSON.parse(localStorage.getItem('voteList'));
    const newVoteList = [
      {
        name,
        path,
        id: name,
        point: 0,
      },
      ...(localStorageVoteList !== null ? localStorageVoteList : []),
    ];
    localStorage.setItem('voteList', JSON.stringify(newVoteList));
  };

  return (
    <section className="addVoteCard">
      <div className="addVoteCard_returnButton">
        <Link to="/list">
          <Icon name="arrow left" />
          Return to list
        </Link>
      </div>
      <div className="addVoteCard_title">
        <h1>Add New Link</h1>
      </div>
      <div className="addVoteCard_form">
        <Form onSubmit={handleSubmit}>
          <label htmlFor="name">
            Link Name:
            <Input
              id="name"
              type="text"
              value={name}
              placeholder="e.g. Alphabet"
              onChange={handleNameChange}
            />
          </label>
          <label htmlFor="path">
            Link URL:
            <Input
              id="path"
              value={path}
              type="text"
              placeholder="e.g. http://abc.xyz"
              onChange={handlePathChange}
            />
          </label>
          <Button type="submit">ADD</Button>
        </Form>
      </div>
    </section>
  );
};

export default AddVoteCardContainer;
