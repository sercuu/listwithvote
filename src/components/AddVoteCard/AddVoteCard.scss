.addVoteCard {
  width: 400px;
  &_returnButton {
    margin: 25px 0;
    a {
      color: #000;
      font: {
        size: 20px;
        weight: bolds;
      }
    }
  }
  &_title {
    margin: 40px 0;
  }
  &_form {
    label {
      display: block;
      margin: 30px 0;
    }
    .input {
      width: 100% !important;
    }
  }
}
