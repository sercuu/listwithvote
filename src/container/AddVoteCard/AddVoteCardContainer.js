import React from 'react';

export const AddVoteCardContainer = () => {
  return <div> helle add</div>;
};

export default AddVoteCardContainer;
