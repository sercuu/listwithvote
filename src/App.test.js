/* eslint-disable no-undef */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AddVoteCard from './components/AddVoteCard/AddVoteCard';

Enzyme.configure({ adapter: new Adapter() });

describe('ControlledForm', () => {
  const component = shallow(<AddVoteCard />);
  it('renders successfully', () => {
    component.find('#name').simulate('change', { target: { name: 'name', value: 'name' } });
    component.find('#path').simulate('change', { target: { name: 'path', value: 'path' } });
    component.find('Button').simulate('click');
    const getLocalStorage = localStorage.getItem('voteList');
    expect(getLocalStorage).toEqual(
      expect.arrayContaining([expect.objectContaining({ name: 'name' })]),
    );
  });
});
