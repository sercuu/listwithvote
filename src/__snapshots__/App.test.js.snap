// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledForm renders successfully 1`] = `ShallowWrapper {}`;
